### PR TITLE
Implemented FT cases 117, 118

### PR DESCRIPTION
**Pull request by AnkitGaurNms**
Implemented FT cases 117, 118

### CODE REVIEW

File: testing/src/test/java/org/motechproject/nms/testing/it/api/KilkariControllerBundleIT.java [hunk 1]
```diff
@@ -773,5 +773,116 @@ public void verifyFT85() throws IOException,
...
+    @Test
+    public void verifyFT117() throws IOException, InterruptedException {
+        /**
+         * To check NMS is able to make available a single message of current
+         * week in inbox with single message per week configuration . when a)
+         * user's MSISDN is subscribed for 48Weeks Pack. b)user's MSISDN is
+         * deactivated for an old subscription of 72Weeks Pack.
+         **/
+        // setup data to remove 2 messages per week configuration for 72 week pack
+        testingService.clearDatabase();
```
Comment (frankhuster) [inline testing/src/test/java/org/motechproject/nms/testing/it/api/KilkariControllerBundleIT.java]: setupData() like 114 takes care to delete the database and create rh & sh.
No need to do the same here.

File: testing/src/test/java/org/motechproject/nms/testing/it/api/KilkariControllerBundleIT.java [hunk 2]
```diff
@@ -773,5 +773,116 @@ public void verifyFT85() throws IOException,
...
+        rh = new RegionHelper(languageDataService, circleDataService,
+                stateDataService, districtDataService);
+
+        sh = new SubscriptionHelper(subscriptionService, subscriberDataService,
+                subscriptionPackDataService, languageDataService,
+                circleDataService, stateDataService, districtDataService);
+
+        deployedServiceDataService.create(new DeployedService(rh.delhiState(),
+                Service.KILKARI));
```
Comment (frankhuster) [inline testing/src/test/java/org/motechproject/nms/testing/it/api/KilkariControllerBundleIT.java]: This is all duplicate code from setupData(). Please remove.

File: testing/src/test/java/org/motechproject/nms/testing/it/api/KilkariControllerBundleIT.java [hunk 3]
```diff
@@ -773,5 +773,116 @@ public void verifyFT85() throws IOException,
...
+    public void verifyFT118() throws IOException, InterruptedException {
+        /**
+         * To check NMS is able to make available a single message of current
+         * week in inbox with single message per week configuration . when a)
+         * user's MSISDN is subscribed for 48Weeks Pack. b)user's MSISDN status
+         * is completed for an old subscription of 72Weeks Pack.
+         **/
+        // setup data to remove 2 messages per week configuration for 72 week
+        // pack
+        testingService.clearDatabase();
```
Comment (frankhuster) [inline testing/src/test/java/org/motechproject/nms/testing/it/api/KilkariControllerBundleIT.java]: duplicate code. setupData() does the same. Please remove

File: testing/src/test/java/org/motechproject/nms/testing/it/utils/SubscriptionHelper.java [hunk 4]
```diff
@@ -67,6 +67,15 @@ public SubscriptionPack pregnancyPack() {
         return subscriptionService.getSubscriptionPack("pregnancyPack");
     }
 
+    public SubscriptionPack pregnancyPack(int messagesPerWeek) {
+        if (subscriptionPackDataService.byName("pregnancyPack") == null) {
+            createSubscriptionPack("pregnancyPack",
+                    SubscriptionPackType.PREGNANCY, PREGNANCY_PACK_WEEKS,
+                    messagesPerWeek);
+        }
+        return subscriptionService.getSubscriptionPack("pregnancyPack");
```
Comment (frankhuster) [inline testing/src/test/java/org/motechproject/nms/testing/it/utils/SubscriptionHelper.java]: What if the existing subscriptionPack does not have the same number of messages per week?

**Comment by AnkitGaurNms** [comment]
Merged branch 'master' into NMS.FT.117.118.
Changed implementation of FT cases 117, 118 as per review comments.

**Comment by srivastavaAbhi** [comment]
test this please

**Comment by frankhuster** [comment]
please merge

**Comment by srivastavaAbhi** [comment]
Tracked with branch NMS.FT.KILKARICONTROLLERIT.
So closing this request.

**Pull request closed without merging**